packages/blade/src/components/ActionList/ActionList.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import Box from '~components/Box';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { getPlatformType, makeAccessible } from '~utils';
+
+type ActionListItemProps = {
+  title: string;
+  value: string;
+  href?: string;
+  /**
+   * Internally passed from ActionList. No need to pass it explicitly
+   */
+  index?: number;
+};
+const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+  const {
+    activeIndex,
+    dropdownBaseId,
+    onOptionClick,
+    selectedIndices,
+    setShouldIgnoreBlur,
+  } = useDropdown();
+
+  const platformType = getPlatformType();
+  const renderOnWebAs = props.href ? 'a' : 'button';
+  const isReactNative = platformType === 'react-native';
+
+  return (
+    <Box
+      as={!isReactNative ? renderOnWebAs : undefined}
+      id={`${dropdownBaseId}-${props.index}`}
+      role="option"
+      data-value={props.value}
+      data-index={props.index}
+      aria-selected={
+        typeof props.index === 'number' ? selectedIndices.includes(props.index) : undefined
+      }
+      onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+        if (typeof props.index === 'number') {
+          onOptionClick(e, props.index);
+        }
+      }}
+      onMouseDown={() => {
+        setShouldIgnoreBlur(true);
+      }}
+      href={props.href}
+      style={{
```

Comment:
styling I am adding in another ActionList PR. don't think so a todo is needed since we definitely have to fix it before we merge all this. Anyway I have a jira ticket to track it.